Resources/doc/suggesters/indexed_suggesters.rst

```diff
@@ -0,0 +1,231 @@
+
+    <?php
+
+    namespace Acme\DemoBundle\Document;
+
+    use ONGR\ElasticsearchBundle\Annotation as ES;
+
+    /**
+     * Product document.
+     *
+     * @ES\Document(type="product")
+     */
+    class Product implements DocumentInterface
+    {
+        /**
+         * @var CompletionSuggesting
+         *
+         * @ES\Suggester\CompletionSuggesterProperty(
+         *  name = "completion_suggesting",
+         *  objectName = "AcmeDemoBundle:CompletionSuggesting",
+         *  index_analyzer = "simple",
+         *  search_analyzer = "simple",
+         *  payloads = false,
+         *  )
+         */
+        public $completionSuggesting;
+    }
+
+As you can see we need to define an object for it, so we can store and retrieve its data objectively.
+In order to do that, we can just use a trait already prepared in ESB.
+
+.. code:: php
+
+    <?php
+
+    namespace Acme\DemoBundle\Document;
+
+    use ONGR\ElasticsearchBundle\Annotation as ES;
+    use ONGR\ElasticsearchBundle\Document\Suggester\CompletionSuggesterInterface;
+    use ONGR\ElasticsearchBundle\Document\Suggester\CompletionSuggesterTrait;
+
+    /**
+     * Suggesting document for testing.
+     *
+     * @ES\Object
+     */
+    class CompletionSuggesting implements CompletionSuggesterInterface
+    {
+        use CompletionSuggesterTrait;
+    }
+
+Context
+~~~~~~~
+
+Context suggester does not only use the `parameters`_ used in completion suggester, but you must also set additional context mapping, into which you’ll store your data. To do this, ESB uses special annotation objects.
+
+Here’s an example:
+
+.. code:: php
+
+    <?php
+
+    namespace Acme\DemoBundle\Document;
+
+    use ONGR\ElasticsearchBundle\Annotation as ES;
+
+    /**
+     * Product document.
+     *
+     * @ES\Document(type="product")
+     */
+    class Product implements DocumentInterface
+    {
+        /**
+         * @var PriceLocationSuggesting
+         *
+         * @ES\Suggester\ContextSuggesterProperty(
+         *   name = "suggestions",
+         *   objectName = "AcmeDemoBundle:PriceLocationSuggesting",
+         *   payloads = true,
+         *   context = {
+         * @ES\Suggester\Context\GeoLocationContext(name="location", precision = "5m", neighbors = true, default = "u33"),
+         * @ES\Suggester\Context\CategoryContext(name="price", default = {"red", "green"}, path = "description")
+         *   }
+         * )
+         */
+        public $contextSuggesting;
+    }
+
+Parameters for geo context can be found
+`here <http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/suggester-context.html#_geo_location_mapping>`__,
+and for category context can be found
+`here <http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/suggester-context.html#_category_mapping>`__.
+
+As you can see it also requires object to store data, here’s an example.
+
+.. note:: Note context parameter name must be ``context`` in order for this mapping to work!
```

Comment:
Note: context parameter name must be ``context`` in order for this mapping to work!